rio_tiler/reader.py

```diff
@@ -373,9 +371,12 @@ def part(
 
         if buffer:
             bounds, height, width = _apply_buffer(buffer, bounds, height, width)
+
             # re-calculate the transform given the new bounds, height and width
             vrt_transform, vrt_width, vrt_height = get_vrt_transform(
-                src_dst, bounds, height, width, dst_crs=dst_crs
+                src_dst,
+                bounds,
+                dst_crs=dst_crs,
             )
```

Comment:
We now construct a VRT aligned with the `reading` bounds but which has the `size` defined by its projected resolution.